Source/JavaScriptCore/builtins/JSIteratorConstructor.js

```diff
@@ -71,3 +71,47 @@ function from(value)
     // 6. Return wrapper.
     return @wrapForValidIteratorCreate(iterator, iteratorNextMethod);
 }
+
+// https://tc39.es/proposal-iterator-sequencing/#sec-iterator.concat
+function concat(/* ...items */)
+{
+    "use strict";
+
+    var count = @argumentCount();
+
+    var openMethods = [];
+    var iterables = [];
+    for (var i = 0; i < count; ++i) {
+        var iterable = arguments[i];
```

Comment:
```suggestion
        var iterable = @argument(i);
```

Reply:
i dont think this works.  every other use of `@arguments()` passes in a constant, and there's lots of places that use `arguments[]`.  when i tried this it kept giving me the first argument over and over, regardless of the value of `i`